src/mpid/ch4/netmod/ucx/ucx_win.h

```diff
@@ -53,11 +54,21 @@ static inline int MPIDI_UCX_Win_allgather(MPIR_Win * win, size_t length,
      mem_map_params.length = size;
      mem_map_params.flags = 0 ;
 
+    if (*base_ptr == NULL)
```

Comment:
Good point. My change just followed the original code. My guess is no, we do not need (if considers raw IB verbs code). Unless UCX has special restriction on this, which should be the reason why the original code used a dummy 1K buffer. I will confirm.